src/test/java/net/datafaker/script/ProvidersDocsGenerator.java

```diff
@@ -0,0 +1,188 @@
+package net.datafaker.script;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ast.comments.CommentsCollection;
+import com.github.javaparser.ast.comments.JavadocComment;
+import com.google.common.base.Strings;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ProvidersDocsGenerator {
+
+    private static final Pattern pattern = Pattern.compile("\\d\\.\\d\\.\\d");
+    private final JavaParser parser = new JavaParser();
+
+    // Specify destination of 'providers.md' file
+    private final String destinationPlaceOfProvidersFile = "src/test/java/net/datafaker/script/providers.md";
+
+    // Need to update this list when adding a new class in 'src.main.java.net.datafaker' which is not Faker.
+    // Store files with '.java' extension in order to avoid unnecessary transformation (substring '.java')
+    private final Set<String> fakersToExcludeFromGeneration = new HashSet<>(Arrays.asList("Faker.java", "CreditCardType.java", "AbstractProvider.java", "FakeCollection.java"));
```

Comment:
In fact it is possible to use reflections lib from here https://github.com/ronmamo/reflections
then just to get all real fakers
```java
Set<Class<?>> subTypes =
  reflections.get(SubTypes.of(SomeType.class).asClass());
```
and no need to do the manual work mentioned in comments

Reply:
does doing that give you access to all the Javadoc? Or do you mean just for this list of Fakers?

Reply:
list of classes which could be used to read source code.

As I understand the initial idea it tries to through all java files within the folder except those mentioned in `fakersToExcludeFromGeneration` which should be constantly updated manually as stated in comments.

so it's better to get all the fakers classes as children of `AbstractProvider` and use this list to read files. Package info could be used to construct real path however it's as one of the ways

Reply:
@snuyanzin Yes, you understood everything correctly. It is not very pleasant to add filenames that are not providers to the set every time. Good idea, never thought of that. But, if I understand correctly, there will be a lot of rewriting, or not?

Reply:
@panilya Look, while I agree with with @snuyanzin , for a simple utility class like this, I wouldn't bother to go this far. The chance of adding a new class to this package isn't that big, and even if it happens, it's a small fix to add it to the list of exclusions here. Personally, I'd just keep it like this, it's already incredibly helpful to be able to generate the correct documentation even once.

Reply:
not really, i can have a closer look at it later today to see if could be done with minimal changes

Reply:
@bodiam I will gladly rewrite, first of all, I try to learn something new from it, and thank you for the feedback. And I always try to do everything as best as possible, so, I will try at least

Reply:
@snuyanzin 
```java
Set<Class<?>> subTypes =
  reflections.get(SubTypes.of(AbstractProvider.class).asClass());
```
This will give a Set of all classes that extends `AbstractProvider`?

Reply:
yes, it should, at least it is stated in their doc

Reply:
It's working now, but Set doesn't guarantee an ordering, so, `providers.md` is unsorted thus looking kind of unreadable. Is it okay to use TreeSet in this case?

Reply:
yes, if you need just lexicographic ordering, otherwise also need a corresponding comparator

Reply:
Absolutely, a TreeSet would be a great choice!